pykafka/simpleconsumer.py

```diff
@@ -462,6 +477,15 @@ def _handle_success(parts):
                     [part for errcode, parts in parts_by_error.iteritems()
                      for part in parts])
 
+            # release all locks to allow fetching
+            for errcode, owned_partitions in parts_by_error.iteritems():
```

Comment:
Not sure, you mean the errored partitions will still be at the offset they were at before the call to `reset_offsets`? That's also true of current master. The thing about raising an exception is that it means the client has to handle it. Since `reset_offsets` is actually called in response to another error condition, this could get messy.